src/m365/pa/commands/app/app-owner-set.spec.ts

```diff
@@ -0,0 +1,370 @@
+      if (opts.url === `https://graph.microsoft.com/v1.0/users?$filter=userPrincipalName eq '${formatting.encodeQueryParameter(validUserName)}'&$select=Id`) {
+        return userResponse;
+      }
+
+      throw 'Invalid request';
+    });
+
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://api.powerapps.com/providers/Microsoft.PowerApps/scopes/admin/environments/${validEnvironmentName}/apps/${validAppName}/modifyAppOwner?api-version=2022-11-01`) {
+        return appOwnerSetResponse;
+      }
+    });
+
+    await command.action(logger, { options: { verbose: true, environmentName: validEnvironmentName, appName: validAppName, userName: validUserName } });
+    assert(postStub.called);
+  });
+
+  it('sets a new Power Apps owner using userId', async () => {
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://api.powerapps.com/providers/Microsoft.PowerApps/scopes/admin/environments/${validEnvironmentName}/apps/${validAppName}/modifyAppOwner?api-version=2022-11-01`) {
+        return appOwnerSetResponse;
+      }
+    });
+
+    await command.action(logger, { options: { environmentName: validEnvironmentName, appName: validAppName, userId: validUserId } });
+    assert(postStub.called);
+  });
+
+  it('sets a new Power Apps owner using userId and sets old owner with role CanView', async () => {
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://api.powerapps.com/providers/Microsoft.PowerApps/scopes/admin/environments/${validEnvironmentName}/apps/${validAppName}/modifyAppOwner?api-version=2022-11-01`) {
+        return appOwnerSetResponse;
+      }
+    });
+
+    await command.action(logger, { options: { environmentName: validEnvironmentName, appName: validAppName, userId: validUserId, roleForOldAppOwner: 'CanView' } });
+    assert(postStub.called);
+  });
+
+  it('sets a new Power Apps owner using userId and sets old owner with role CanEdit', async () => {
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://api.powerapps.com/providers/Microsoft.PowerApps/scopes/admin/environments/${validEnvironmentName}/apps/${validAppName}/modifyAppOwner?api-version=2022-11-01`) {
+        return appOwnerSetResponse;
+      }
+    });
+
+    await command.action(logger, { options: { environmentName: validEnvironmentName, appName: validAppName, userId: validUserId, roleForOldAppOwner: 'CanEdit' } });
+    assert(postStub.called);
+  });
+
+  it('correctly handles API OData error', async () => {
+    const errorMessage = `The specified user with user id ${validUserId} does not exist.`;
+    sinon.stub(request, 'post').rejects({
+      error: {
+        'odata.error': {
+          code: '-1, InvalidOperationException',
+          message: {
+            value: errorMessage
+          }
```

Comment:
API error looks like:

"error": {
    "error": {
      "message": "Nope, that doesn't work!"
   }
}